### PR TITLE
Optimise getPostObjectName func

### PR DESCRIPTION
**Pull request by caoyingjunz**
Ⅰ. Describe what this PR does

Ⅱ. Does this pull request fix one issue?

Ⅲ. List the added test cases (unit test/integration test) if any, please explain if no tests are needed.

Ⅳ. Describe how to verify it

Ⅴ. Special notes for reviews

### CODE REVIEW

File: pkg/apiserver/middlewares/audit/audit.go [hunk 1]
```diff
@@ -340,40 +340,31 @@ func (r responseBodyWriter) WriteString(s string) (n int, err error) {
...
 	}
-	nameStr, ok := name.(string)
-	if !ok {
-		clog.Warn("[audit] convert name to string failed")
-		return ""
+
+	return obj.Metadata.Name
+}
+
+func ShouldBindWith(c *gin.Context, v interface{}) error {
```
Comment (weilaaa) [inline pkg/apiserver/middlewares/audit/audit.go:358]: It seems there is no need to export this func.

File: pkg/apiserver/middlewares/audit/audit.go [hunk 2]
```diff
@@ -340,40 +340,31 @@ func (r responseBodyWriter) WriteString(s string) (n int, err error) {
...
-	}
-	// put request body back
-	c.Request.Body = ioutil.NopCloser(bytes.NewBuffer(data))
+	obj := struct {
+		Metadata struct {
+			Name string `json:"name"`
+		} `json:"metadata"`
+	}{}
+
 	// get resource name from metadata.name
```
Comment (weilaaa) [inline pkg/apiserver/middlewares/audit/audit.go:349]: Please modify the meaning of this comment to express the meaning of the function.

File: pkg/apiserver/middlewares/audit/audit.go [hunk 3]
```diff
@@ -340,40 +340,31 @@ func (r responseBodyWriter) WriteString(s string) (n int, err error) {
...
-	}
-	metadataMap, ok := metadata.(map[string]interface{})
-	if !ok {
-		clog.Warn("[audit] convert metadata to map failed")
-		return ""
-	}
-	name, exist := metadataMap["name"]
-	if !exist {
+	if err := ShouldBindWith(c, &obj); err != nil {
 		clog.Warn("[audit] the resource metadata.name is nil")
```
Comment (weilaaa) [inline pkg/apiserver/middlewares/audit/audit.go:351]: We should not ignore error here.

File: pkg/apiserver/middlewares/audit/audit.go [hunk 4]
```diff
@@ -340,40 +340,31 @@ func (r responseBodyWriter) WriteString(s string) (n int, err error) {
...
 func getPostObjectName(c *gin.Context) string {
-	// get request body
-	data, err := c.GetRawData()
-	if err != nil {
-		clog.Warn("[audit] get raw data err: %s", err.Error())
-		return ""
-	}
-	// put request body back
-	c.Request.Body = ioutil.NopCloser(bytes.NewBuffer(data))
+	obj := struct {
```
Comment (weilaaa) [inline pkg/apiserver/middlewares/audit/audit.go:343]: How about using `unstructured.Unstructured` with couple of `xxx.GetName()` and `xxx.UnmarshalJSON()`, I just leave it for @caoyingjunz.

Comment (caoyingjunz) [reply pkg/apiserver/middlewares/audit/audit.go:343]: > How about using `unstructured.Unstructured` with couple of `xxx.GetName()` and `xxx.UnmarshalJSON()`, I just leave it for @caoyingjunz.

@weilaaa done, thanks for advice

Comment (caoyingjunz) [reply pkg/apiserver/middlewares/audit/audit.go:343]: @weilaaa please help review https://github.com/kubecube-io/KubeCube/pull/264

**Pull request closed without merging**